test/support/assert_api_response.rb

```diff
@@ -30,6 +30,53 @@ def assert_api_request(method, path, params = {})
 
     assert_mock http
   end
+
+  # Asserts that the given resource can use auto-pagination.
+  #
+  # path     - API path (without `/api/v1` prefix, eg: "/messages")
+  # resource - ReamazeAPI resource name (eg: `:messages`)
+  #
+  # Example
+  #
+  #   assert_pagination "/conversations", :conversations
+  def assert_pagination(path, resource)
```

Comment:
@meesterdude The page size doesn't really matter, we just need to know how many pages so we can check if we still have pages to fetch.

Say we have 150 conversations. The API returns a response like this:

```json
{
  "page_size": 30,
  "page_count": 5,
  "total_count": 150,
  "conversations": [
    # conversations 1-30
  ]
}
```

You would need to hit `/api/v1/conversations?page=2`, through 5 to get all 150 messages.

I'm trying to test that a paginated call returns the full list, so for example that the `conversations` key in the above payload contained all 150 conversations and not just 1-30 from the one API call.

The stubbing simulates 3 requests with a page size of 2 (because the page size doesn't really matter, we just need to know how many pages there are) and returns the keys needed for pagination to work, for example:

```
# /api/v1/conversations?page=1
{
  "page_count": 3,
  "conversations": [
    { slug: "conversations-p1-1" },
    { slug: "conversations-p1-2" }
  ]
}

# /api/v1/conversations?page=2
{
  "page_count": 3,
  "conversations": [
    { slug: "conversations-p2-1" },
    { slug: "conversations-p2-2" }
  ]
}

# /api/v1/conversations?page=3
{
  "page_count": 3,
  "conversations": [
    { slug: "conversations-p3-1" }
  ]
}
```

The expectation is that I can run `client.conversations.all(auto_paginate: true)`, and get a response with all 5 of the conversations:

```ruby
client.conversations.all(auto_paginate: true)
# {
#   success: true,
#   payload: {
#     page_count
#     page_size: 2,
#     page_count: 3,
#     total_count: 5,
#     conversations: [
#       { slug: "conversations-p1-1" },
#       { slug: "conversations-p1-2" },
#       { slug: "conversations-p2-1" },
#       { slug: "conversations-p2-2" },
#       { slug: "conversations-p3-1" }
#     ]
#   }
# }